client/src/components/Form/TextField.tsx

```diff
@@ -72,6 +72,9 @@ const parsers: ParsersMap = {
 
     return typeof value === 'number' ? clamp(value, min, max) : value;
   },
+  text: (event) => {
```

Comment:
I think this was a goof on my part before. There is a bug in this component that if you use a `text` input, the form state isn't updated properly, but rather than fixing it with a `parser`, I think we just make this the default behavior. I'd recommend changing `onChange` to look like this instead:

```ts
onChange={(event) => {
  if (parser) {
    return setValue(parser(event, omit(props, ['type'])));
  }

  setValue(event.target.value);
}}
```

The `parser` was really mean to be for type casting the raw value to something else (i.e. a number), but leaving it as plain text in the default case seems sensible to me.

Reply:
Makes sense - made that update and still allowed the caller to pass in its own `onChange` that also gets called. Updated in [979b8f7](https://github.com/apollographql/spotify-showcase/pull/87/commits/979b8f7d47b9d8a02e72782b9353de95333680a6).